Modules/Quotas/test/Quotas.Application.Tests/Tests/Quotas/CreateQuotaForIdentity/HandlerTests.cs

```diff
@@ -62,8 +62,10 @@ public async Task Create_quota_with_invalid_metric_key_throws_domain_exception()
         var handler = CreateHandler(identitiesRepository, metricsRepository);
 
         // Act
-        var exception = await Assert.ThrowsAsync<DomainException>(async () => await handler.Handle(command, CancellationToken.None));
-        exception.Message.Should().Contain("The given metric key is not supported.");
+        var act = async () => await handler.Handle(command, CancellationToken.None);
+
+        // Assert
+        var exception = (await act.Should().ThrowAsync<DomainException>()).And;
```

Comment:
There is a `AwaitAndThrowAsync` method we implemented which allows you to remove the `await`